live/js.go

```diff
@@ -0,0 +1,198 @@
+	TransitionClass string
+	// StartClass is the CSS class(es) that apply at the start of a transition.
+	StartClass string
+	// EndClass is the CSS class(es) that apply at the end of a transition.
+	EndClass string
+}
+
+type HideOpts struct {
+	// To is the DOM selector of the element to hide, or empty to target the interacted element.
+	To string
+	// Transition to apply, if any.
+	Transition *Transition
+	// Time is the duration in milliseconds of the transition, if present; defaults to DefaultTransitionDuration_ms if 0.
+	Time int32
+}
+
+func (o *HideOpts) args() map[string]any {
+	var to *string
+	time := DefaultTransitionDuration_ms
+	transition := &Transition{}
+	if o != nil {
+		to = &o.To
+		if o.Time != 0 {
+			time = o.Time
+		}
+		if o.Transition != nil {
+			transition = o.Transition
+		}
+	}
+	return map[string]any{
+		"time":       time,
+		"to":         to,
+		"transition": transition.phx(),
+	}
+}
+
+type ShowOpts struct {
+	// To is the DOM selector of the element to show, or empty to target the interacted element.
+	To string
+	// Transition to apply, if any.
+	Transition *Transition
+	// Time is the duration in milliseconds of the transition, if present; defaults to DefaultTransitionDuration_ms if 0.
+	Time int32
+	// Display is the CSS display value to set when showing; defaults to "block".
+	Display string
+}
+
+func (o *ShowOpts) args() map[string]any {
+	var to *string
+	time := DefaultTransitionDuration_ms
+	transition := &Transition{}
+	display := "block"
+	if o != nil {
+		to = &o.To
+		if o.Time != 0 {
+			time = o.Time
+		}
+		if o.Transition != nil {
+			transition = o.Transition
+		}
+		if o.Display != "" {
+			display = o.Display
+		}
+	}
+	return map[string]any{
+		"time":       time,
+		"to":         to,
+		"transition": transition.phx(),
+		"display":    display,
+	}
+}
+
+type ToggleOpts struct {
+	// To is the DOM selector of the element to toggle visibility of, or empty to target the interacted element.
+	To string
+	// In is the transition to apply when showing the element.
+	In *Transition
+	// Out is the transition to apply when hiding the element.
+	Out *Transition
+	// Time is the duration in milliseconds of the relevant transition, if present; defaults to DefaultTransitionDuration_ms if 0.
+	Time int32
+	// Display is the CSS display value to set when showing; defaults to "block".
+	Display string
+}
+
+func (o *ToggleOpts) args() map[string]any {
+	var to *string
+	time := DefaultTransitionDuration_ms
+	in := &Transition{}
+	out := &Transition{}
+	display := "block"
+	if o != nil {
+		to = &o.To
+		if o.Time != 0 {
+			time = o.Time
+		}
+		if o.In != nil {
+			in = o.In
+		}
+		if o.Out != nil {
+			out = o.Out
+		}
+		if o.Display != "" {
+			display = o.Display
+		}
+	}
+	return map[string]any{
+		"time":    time,
+		"to":      to,
+		"ins":     in.phx(),
+		"outs":    out.phx(),
+		"display": display,
+	}
+}
+
+func phxClasses(class string) []string {
+	if class == "" {
+		return []string{}
```

Comment:
Does it need to be an empty slice, or can it be nil?

Reply:
I note this above, but the phoenix js client expects an empty js array as representing "no transition class"